src/main/java/org/jenkinsci/plugins/workflow/steps/RetryStepExecution.java

```diff
@@ -61,6 +65,18 @@ public void onSuccess(StepContext context, Object result) {
         @Override
         public void onFailure(StepContext context, Throwable t) {
             try {
+                Run run = context.get(Run.class);
+                if (run != null && t instanceof FlowInterruptedException) {
+                    InterruptedBuildAction action = run.getAction(InterruptedBuildAction.class);
+                    if (action != null) {
+                        for (CauseOfInterruption cause : action.getCauses()) {
```

Comment:
I tried that in 3c2bba9 but the exception didn't contain any causes to check. See https://jenkins.ci.cloudbees.com/job/plugins/job/workflow-basic-steps-plugin/org.jenkins-ci.plugins.workflow$workflow-basic-steps/187/

So reverted that change in 3c9dfec